mythril/laser/ethereum/natives.py

```diff
@@ -53,30 +57,35 @@ def ecrecover(data):
     return o
 
 
-def sha256(data):
+def sha256(data: Union[bytes, str]) -> bytes:
     try:
         data = bytes(data)
     except TypeError:
         raise NativeContractException
     return hashlib.sha256(data).digest()
 
 
-def ripemd160(data):
+def ripemd160(data: Union[bytes, str]) -> bytes:
     try:
         data = bytes(data)
     except TypeError:
         raise NativeContractException
-    return 12 * [0] + [i for i in hashlib.new("ripemd160", data).digest()]
+    digest = hashlib.new("ripemd160", data).digest()
+    padded = 12 * [0] + list(digest)
+    return bytes(padded)
 
 
-def identity(data):
+def identity(data: Union[bytes, str]) -> bytes:
+    try:
+        data = bytes(data)
```

Comment:
Doesn't this already create some kind of copy of data?

Reply:
In any case, it seems like bytes is an immutable type, so copying wouldn't make sense anyway right?

Reply:
But right now the copy is unnecessary right?

Reply:
yep